File: src/main/java/preferences/closedloopfueling/ClosedLoopFuelingLogPreferences.java
package preferences.closedloopfueling;

import javax.swing.filechooser.FileSystemView;
import java.io.File;
import java.util.prefs.Preferences;

public class ClosedLoopFuelingLogPreferences {

    private static final String MIN_THROTTLE_ANGLE_PREFERENCE = "min_throttle_angle_preference";
    private static final String MIN_RPM_PREFERENCE = "min_rpm_preference";
    private static final String MAX_VOLTAGE_DT_PREFERENCE = "max_voltage_dt_preference";

    private static final String LAST_USED_DIRECTORY_KEY = "closed_loop_last_used_directory_key";

    private static final Preferences prefs = Preferences.userNodeForPackage(ClosedLoopFuelingLogPreferences.class);

    public static double getMinThrottleAnglePreference() {
        return Double.parseDouble(prefs.get(MIN_THROTTLE_ANGLE_PREFERENCE, String.valueOf(0)));
    }

    public static void setMinThrottleAnglePreference(double minThrottleAngle) {
        prefs.put(MIN_THROTTLE_ANGLE_PREFERENCE, String.valueOf(minThrottleAngle));
    }

    public static double getMinRpmPreference() {
        return Double.parseDouble(prefs.get(MIN_RPM_PREFERENCE, String.valueOf(0)));
    }

    public static void setMinRpmPreference(double minRpm) {
        prefs.put(MIN_RPM_PREFERENCE, String.valueOf(minRpm));
    }

    public static double getMaxVoltageDtPreference() {
        return Double.parseDouble(prefs.get(MAX_VOLTAGE_DT_PREFERENCE, String.valueOf(1)));
    }

    public static void setMaxVoltageDtPreference(double maxDerivative) {
        prefs.put(MAX_VOLTAGE_DT_PREFERENCE, String.valueOf(maxDerivative));
    }

    public static File getDirectory() {
        return new File(prefs.get(LAST_USED_DIRECTORY_KEY, FileSystemView.getFileSystemView().getHomeDirectory().getAbsolutePath()));
    }

    public static void setDirectory(File file) {
        prefs.put(LAST_USED_DIRECTORY_KEY, file.getAbsolutePath());
    }
}

File: src/main/kotlin/data/preferences/closedloopfueling/ClosedLoopFuelingLogPreferences.kt
package data.preferences.closedloopfueling

import java.util.prefs.Preferences

object ClosedLoopFuelingLogPreferences {
    private val prefs = Preferences.userNodeForPackage(ClosedLoopFuelingLogPreferences::class.java)

    var minThrottleAngle: Double
        get() = prefs.get("min_throttle_angle_preference", "0").toDouble()
        set(value) = prefs.put("min_throttle_angle_preference", value.toString())

    var minRpm: Double
        get() = prefs.get("min_rpm_preference", "0").toDouble()
        set(value) = prefs.put("min_rpm_preference", value.toString())

    var maxDerivative: Double
        get() = prefs.get("max_derivative_preference", "50").toDouble()
        set(value) = prefs.put("max_derivative_preference", value.toString())

    var lastDirectory: String
        get() = prefs.get("last_directory_preference", "")
        set(value) = prefs.put("last_directory_preference", value)
}
